### ISSUE TITLE
How would I layout subcollections in the mock database?

### ISSUE THREAD

**Issue by AverageHelper** [Upstatement/firestore-jest-mock#22]
# Summary
I'm trying to mock Firestore data that's a few layers deep in the collection/document tree. In my testing so far, the mocked `get()` function doesn't resolve for more than 5 seconds, and then Jest times out.

## Relevant information
My mock tree is laid out like so:
```(JavaScript)
mockFirebase({
  database: {
    accounts: [  // collection, right?
      {
        id: testAcctId,
        conversations: [  // subcollection, right?
            { id: testConversationId, participants: ["someone", "someone-else"] }
        ],
        userPermissions: [  // another subcollection, right?
          { id: "someone", level: "normal" },
          { id: "someone-else", level: "admin" }
        ]
      }
    ]
  }
});
```

My coverage report says the test function stops at this call:
```(JavaScript)
const { firestore } = require("../my/firestoreSetupFile"); // We reexport require("firebase-admin") from here
const userPermissionsSnapshot = await firestore
      .collection("accounts")
      .doc(accountId)
      .collection("userPermissions")
      .doc(userId)
      .get();  // We hang here ~5 seconds before timeout. What's up with that?
```

The way I understand it, and based on the README, it seems that `firestore-jest-mock` interprets arrays of objects with the `id` attribute to denote subcollections of documents.  Am I incorrect in that assumption?  And could it be that it only assumes that collections are at the top level?  If not, do I have the syntax right to mock a database with 2+ layers of subcollections?  I have more, but it is at this point that my tests fail to run properly, and I'm not a little confused.

### Environment
I'm using Jest 25.5.3

**Comment by AverageHelper** [comment]
Actually, there may in fact be something else going wrong... my code will also freeze in the same way if I just call one collection deep:
```(JavaScript)
console.log(
    await firestore
      .collection("accounts")
      .doc(accountId)
      .get()  // We hang here, then quit...
);
```

**Comment by AverageHelper** [comment]
It turns out that `firestore` isn't being mocked at all!  It may be because I import my setup file before I call `mockFirestore`. So, I remove that, and I remember why I did it in the first place...

Kinda jank, but some of my helper files use `firestore.FieldValue`, which doesn't seem to be defined on the Firestore mock. At the very least, I find these errors in the console:
```
Cannot read property 'serverTimestamp' of undefined
firebase.firestore.FieldValue.serverTimestamp(),
                              ^
```

It occurs to me that this is an entirely different issue that I should've figured out before starting up this issue. I'll close this and write up a more relevant one with my findings once it's a bit less late at night for me.

**Comment by sbatson5** [comment]
This was a fun morning read 😄 

You're right about `FieldValue` -- that is not something this library is currently mocking.  We are not close to 100% parity with firestore (it's really just collections we are handling and a few auth calls).  I'd like to add more to make this library more feature complete.  PRs are welcome or if you want to open an issue documenting how you use FieldValue, I can set aside some time soon to take a look!

**Comment by AverageHelper** [comment]
Thank you.  I started another issue at #23.
